packages/connection/src/class-plugin-storage.php

```diff
@@ -116,11 +137,55 @@ public static function delete( $slug ) {
 	 * @return bool|WP_Error
 	 */
 	private static function ensure_configured() {
-		if ( class_exists( Config::class ) && method_exists( Config::class, 'is_configured' ) && ! Config::is_configured() ) {
```

Comment:
I like your approach on checking if the class is configured.
I guess we could also remove `Config::on_configured()` and `Config::is_configured()`, as well as static variables they use (`is_configuration_needed` and `is_configured`). They aren't used by anything else.

There are also a few methods with PHPDoc like `the method cannot be called until Jetpack Config has been run`. These statements are no longer valid, can you please adjust them too?

Reply:
Please see if https://github.com/Automattic/jetpack/pull/15726/commits/244713998643cf8a9e1092dd1599a78ecdb2c087 looks good

Reply:
I'm hesitant about this and want to make sure I understand it. The `configured` code's intent is to ensure that the config functions only run once. I'm not certain about the impact of this.

We can clean up the class exist check though by ensuring that the Connection package relies on the Config package, then the autoloader would take care of it and we wouldn't need to check for the class/method.

Reply:
@kraftbj I think you looked at an old diff there. I changed this check and how we handle it.

The idea here is actually to assure any method to get the registered plugins are only called after the Config class was initialized (which happens on `plugins_loaded` priority 2)